### PR TITLE
Feat/RTENU-192 filelist api

### PR DESCRIPTION
**Pull request by saneDG**
- Add endpoints for folder components
  - createFolder
  - updateFolder
  - deleteFolder
  - getComponents
    - returns list of pages every `categoryComponent` item. These are used to render "folder components" in frontend

extra stuff
- add `devLog` function that can be used to log freely in development

### CODE REVIEW

File: packages/server/lambdas/database/components/get-coponents.ts
Comment (NinaDang97) [inline packages/server/lambdas/database/components/get-coponents.ts:6]: singular `getComponent`? `findFirst` returns 1 record

Comment (NinaDang97) [inline packages/server/lambdas/database/components/get-coponents.ts:6]: Where is this function `getComponent` used? Do we need it?

File: packages/server/lambdas/alfresco/list-components.ts
Comment (NinaDang97) [inline packages/server/lambdas/alfresco/list-components.ts:33]: I assume this line does what `get-components.ts` does

File: packages/server/lambdas/database/components/get-coponents.ts
Comment (NinaDang97) [reply packages/server/lambdas/database/components/get-coponents.ts:6]: File name has typo, missing `m`

File: packages/server/lambdas/alfresco/delete-folder.ts
Comment (NinaDang97) [inline packages/server/lambdas/alfresco/delete-folder.ts:67]: Should we have conditional check if `deleteFolder` succeeded in Alfresco, so can continue with `deleteComponent` in database? In case, delete in Alfresco fails could lead to data corruption.

File: packages/server/lambdas/database/components/update-node-component.ts
Comment (NinaDang97) [inline packages/server/lambdas/database/components/update-node-component.ts:8]: In what case do we use this (in future)?

File: packages/server/lambdas/alfresco/list-components.ts
Comment (NinaDang97) [inline packages/server/lambdas/alfresco/list-components.ts:20]: log.info missing?

File: lib/rataextra-backend.ts
Comment (mehtis) [inline lib/rataextra-backend.ts:360]: `folders` while others are `folder`

File: packages/server/lambdas/alfresco/create-folder.ts
Comment (mehtis) [inline packages/server/lambdas/alfresco/create-folder.ts:74]: This should be auditlog.info. Also a small thing: Start `created` with big C.

File: packages/server/lambdas/database/components/create-node-component.ts
Comment (mehtis) [inline packages/server/lambdas/database/components/create-node-component.ts:6]: Commented out code.

**Review by mehtis — CHANGES_REQUESTED** [review]

File: packages/server/lambdas/alfresco/delete-folder.ts
Comment (saneDG) [reply packages/server/lambdas/alfresco/delete-folder.ts]: This one is now fixed

File: packages/server/lambdas/alfresco/list-components.ts
Comment (mehtis) [inline packages/server/lambdas/alfresco/list-components.ts:12]: Shouldn't this be the other way around now? It was with s but that was removed.

**Review by mehtis — APPROVED** [review]
Aside from a very small remark, LGTM. Do get approval from @NinaDang97 as well before merging.

File: packages/server/lambdas/alfresco/list-components.ts
Comment (NinaDang97) [reply packages/server/lambdas/alfresco/list-components.ts:12]: Agree, `/folders` is better

Comment (mehtis) [reply packages/server/lambdas/alfresco/list-components.ts:12]: What I meant was that this was changed to be without s elsewhere, but was changed to contain s here, which was inconsistent.

File: packages/server/lambdas/alfresco/delete-folder.ts
Comment (NinaDang97) [inline packages/server/lambdas/alfresco/delete-folder.ts:73]: @mehtis Is it possible to implement transaction rollback if `deleteComponent` fails in this case?  Deleted folder in Alfresco should be restored. Same with create folder (again data corruption)

Comment (saneDG) [reply packages/server/lambdas/alfresco/delete-folder.ts:73]: This feels something we should implement if we ever encounter such problem.

At least we are sure that folder is deleted from alfresco before we delete it from our database.

**Review by NinaDang97 — APPROVED** [review]

**Pull request merged**
